guarddog/scanners/pypi_package_scanner.py

```diff
@@ -38,31 +38,29 @@ def download_package(self, package_name, directory, version=None) -> str:
         if version is None:
             version = data["info"]["version"]
 
-        if version in releases:
-            files = releases[version]
-
-            url = None
-            file_extension = None
-
-            for file in files:
-                # Store url to compressed package and appropriate file extension
-                if file["filename"].endswith(".tar.gz"):
-                    url = file["url"]
-                    file_extension = ".tar.gz"
-
-                if file["filename"].endswith(".egg") or file["filename"].endswith(".whl") \
-                        or file["filename"].endswith(".zip"):
-                    url = file["url"]
-                    file_extension = ".zip"
-
-            if url and file_extension:
-                # Path to compressed package
-                zippath = os.path.join(directory, package_name + file_extension)
-                unzippedpath = zippath.removesuffix(file_extension)
-
-                self.download_compressed(url, zippath, unzippedpath)
-                return unzippedpath
-            else:
-                raise Exception(f"Compressed file for {package_name} does not exist on PyPI.")
-        else:
-            raise Exception("Version " + version + " for package " + package_name + " doesn't exist.")
+        if version not in releases:
+            raise Exception(f"Version {version} for package {package_name} doesn't exist.")
+
+        files = releases[version]
+        url = None
+        file_extension = None
+
+        for file in files:
+            # Store url to compressed package and appropriate file extension
+            if file["filename"].endswith(".tar.gz"):
+                url = file["url"]
+                file_extension = ".tar.gz"
+
+            if any(file["filename"].endswith(ext) for ext in (".egg", ".whl", ".zip")):
+                url = file["url"]
+                file_extension = ".zip"
+
+        if not url and not file_extension:
```

Comment:
I think the opposite of `if url and file_extension` should be `if not url or not file_extension`, right? (instead of `and`)

Reply:
I don't think so since `if url and file_extension` means that both variables need to be set, therefore the exact opposite would be `if not url and not file_extension` or we could also do `if not (url and file_extension)`. Your condition checks only if one of the variables are not set

Reply:
ahh, u guys are right